contrib/profiler/profiler.py

```diff
@@ -0,0 +1,364 @@
+           sum(cpu_when_gpu_low) / length_gpu_low,
+           sum(mem_when_gpu_low) / length_gpu_low,
+           sum(disk_read_when_gpu_low) / length_gpu_low))
+    return [max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+            sum(cpu_when_gpu_low) / length_gpu_low,
+            sum(mem_when_gpu_low) / length_gpu_low,
+            sum(disk_read_when_gpu_low) / length_gpu_low]
+
+
+def start_sample(container_id, period, one_duration, dir, gpu_id, *container_pid):
+    if not os.path.exists('./' + dir):
+        os.mkdir(dir)
+    realtime_log = csv.writer(open('./' + dir + '/log_result.csv', 'w'))  # , newline=''))
+    analyze_log = csv.writer(open('./' + dir + '/analyze_result.csv', 'w'))  # , newline=''))
+
+    str_write_realtime = ['cpu_usage', 'mem_used', 'IO_read', 'IO_write', 'network_receive', 'network_transmit']
+    for i in range(len(gpu_id)):
+        str_write_realtime.append('gpu_usage_' + str(gpu_id[i]))
+        str_write_realtime.append('gpu_mem_used_' + str(gpu_id[i]))
+    realtime_log.writerow(str_write_realtime)
+
+    analyze_log.writerow(['max_gpu', 'avg_gpu', 'max_cpu', 'avg_cpu', 'max_mem', 'avg_mem', 'max_read', 'sum_read',
+                          'avg_cpu_gpu_low', 'avg_mem_gpu_low', 'avg_IO_gpu_low'])
+    nv.nvmlInit()
+    sample_list = list()
+    container_cpu_filelist = list()
+    container_mem_filelist = list()
+    container_blk_filelist = list()
+    # container_net_file = ''
+
+    print(
+        'max_gpu\tavg_gpu\tmax_cpu\tavg_cpu\tmax_memory\tavg_memory\tmax_read\ttotal_read\tavg_cpu_when_gpu_low'
+        '\tavg_mem_when_gpu_low\tavg_io_when_gpu_low')
+
+    if container_pid:
+        container_cpu_filelist = glob.glob('/sys/fs/cgroup/cpuacct/docker/' + str(container_id) + '*/cpuacct.stat')
+        container_mem_filelist = glob.glob(
+            '/sys/fs/cgroup/memory/docker/' + str(container_id) + '*/memory.usage_in_bytes')
+        container_blk_filelist = glob.glob(
+            '/sys/fs/cgroup/blkio/docker/' + str(container_id) + '*/blkio.throttle.io_service_bytes')
+        container_net_file = '/proc/' + str(container_pid[0]) + '/net/dev'
+    else:
+        container_cpu_filelist.append('/sys/fs/cgroup/cpuacct/cpuacct.stat')
+        container_mem_filelist.append('/sys/fs/cgroup/memory/memory.usage_in_bytes')
+        container_blk_filelist.append('/sys/fs/cgroup/blkio/blkio.throttle.io_service_bytes')
+        container_net_file = '/proc/net/dev'
+    while True:
+        # 1st info about I/O and network
+        read_bytes1 = get_disk_read_bytes(container_blk_filelist)
+        write_bytes1 = get_disk_write_bytes(container_blk_filelist)
+        [network_receive1, network_transmit1] = get_network_bytes(container_net_file)
+
+        # CPU usage will cost time('period') running
+        cpu_usage = get_cpu_percent(container_cpu_filelist, period)[-1]
```

Comment:
Can we move the `time.sleep(period)`  from `get_cpu_percent` to the while loop. Then other collection will not depends on `get_cpu_percent` function.
I know we need to collect cpu usage twice to get the rate. We can set a `initialized` flag to solve it.